skimage/feature/tests/test_blob.py

```diff
@@ -246,11 +246,11 @@ def test_blob_log_exclude_border():
     b = blobs[0]
     assert b[0] == b[1] == 5, "blob should be 5 px from x and y borders"
 
-    blobs = blob_dog(
+    blobs = blob_log(
         img,
         min_sigma=1.5,
         max_sigma=5,
-        exclude_border=5
+        exclude_border=6,
```

Comment:
humm why was the test passing before then?

Reply:
blob_* methods takes the input image of dimensionality N and applies a gaussian filter with 7 different sigmas, and makes an image of dimensionality N+1.  Then it applies peak_local_max across that composite image.  however, since `exclude_border` was previously _not_ anistropic and was actually ignored if a footprint was provided, it ended up zeroing out a bunch of the gaussian images.

example:

- input image is (500,500)
- blob_* takes the image and applies gaussian filter with varying sigmas, and produces a (7, 500, 500) image.
- previous behavior of exclude_border + footprint erroneously zeroes [0, :, :], [1, :, :], [5, :, :], and [6, :, :].

This resulted in `blob_log` finding no spots, not as a result of removing the border, but as a result of erroneously zeroing out data.

Reply:
@ttung brilliant explanation! 😀